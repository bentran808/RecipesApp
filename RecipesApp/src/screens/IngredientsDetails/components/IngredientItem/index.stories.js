import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import { ingredientsDetails } from 'mocks';
import React from 'react';
import CenterView from 'storybook/stories/CenterView';
import IngredientItem from '.';

storiesOf('Ingredient Item', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('default', () => (
    <IngredientItem item={ingredientsDetails} onPressIngredient={action('clicked-item')} />
  ));
