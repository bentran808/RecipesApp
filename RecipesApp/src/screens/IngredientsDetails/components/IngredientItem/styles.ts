import { StyleSheet } from 'react-native';
import { width } from 'theme';

const numColumns = 3;
// item size
const RECIPE_ITEM_HEIGHT = 100;
const RECIPE_ITEM_OFFSET = 10;
const RECIPE_ITEM_MARGIN = RECIPE_ITEM_OFFSET * 2;
const RECIPE_ITEM_WIDTH = (width - RECIPE_ITEM_MARGIN) / numColumns - RECIPE_ITEM_OFFSET;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginLeft: RECIPE_ITEM_OFFSET,
    marginBottom: RECIPE_ITEM_OFFSET,
    marginTop: 30,
    width: RECIPE_ITEM_WIDTH,
    height: RECIPE_ITEM_HEIGHT + 60
  },
  title: {
    margin: 10,
    marginBottom: 5,
    color: 'black',
    fontSize: 13,
    textAlign: 'center'
  },
  photo: {
    width: RECIPE_ITEM_WIDTH,
    height: RECIPE_ITEM_HEIGHT,
    borderRadius: 60
  },
  quantity: {
    color: 'grey'
  }
});

export default styles;
