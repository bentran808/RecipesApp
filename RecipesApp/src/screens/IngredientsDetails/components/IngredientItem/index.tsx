import React from 'react';
import isEqual from 'react-fast-compare';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import styles from './styles';

type Props = {
  item: IngredientsDetails;
  onPressIngredient: () => void;
};

const IngredientItem = ({ item, onPressIngredient }: Props) => {
  return (
    <TouchableOpacity onPress={onPressIngredient}>
      <View style={styles.container}>
        <Image style={styles.photo} source={{ uri: item[0].photo_url }} />
        <Text style={styles.title}>{item[0].name}</Text>
        <Text style={styles.quantity}>{item[1]}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default React.memo(IngredientItem, isEqual);
