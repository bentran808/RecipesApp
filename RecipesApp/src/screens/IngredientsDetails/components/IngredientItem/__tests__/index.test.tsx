import { ingredientsDetails } from 'mocks';
import renderer from 'react-test-renderer';
import IngredientItem from '..';

describe('Ingredient Item Component', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<IngredientItem item={ingredientsDetails} onPressIngredient={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
