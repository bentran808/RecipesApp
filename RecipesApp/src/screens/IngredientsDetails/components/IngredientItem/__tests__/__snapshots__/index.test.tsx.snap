// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredient Item Component should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 160,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginTop": 30,
        "width": 233.33333333333334,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://ak7.picdn.net/shutterstock/videos/27252067/thumb/11.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 60,
          "height": 100,
          "width": 233.33333333333334,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "margin": 10,
          "marginBottom": 5,
          "textAlign": "center",
        }
      }
    >
      Oil
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
        }
      }
    >
      200ml
    </Text>
  </View>
</View>
`;
